### PR TITLE
[typing] allow Sequence inputs in several jax.numpy functions

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

**Review by hawkinsp — APPROVED** [review]
I'm curious: did these cause issues for users?

**Comment by jakevdp** [comment]
I noticed one place where we added an ignore statement due ot a user passing a list to the axis argument of `squeeze`. When I saw that, I put this change on my TODO list.
